### PR TITLE
Sign quote UI in search results

### PR DESCRIPTION
**Pull request by alebedev**
# Jira Issue: [CLX-XXX]

## What?
Quote search resutls
- Sign Quote UI (WIP)
- Debug Info

## Why?
Part of IEX Empowerment, allow to sign buggy quotes from comparison sites

## If applicable, rooms for improvement?
- 

## Optional checklist
- [ ] Codescouted
- [ ] Unit tests written
- [ ] Tested locally

### CODE REVIEW

**Comment by alebedev** [comment]
Current dependencies on/for this PR:
* master
  * **PR #1064** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/hedgehog/1064" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/hedgehog/1064?utm_source=stack-comment).

File: package.json [hunk 1]
```diff
@@ -100,7 +100,7 @@
     "react-dom": "18.2.0",
     "react-dropzone": "^10.1.10",
     "react-gtm-module": "^2.0.11",
-    "react-hook-form": "^7.28.1",
+    "react-hook-form": "^7.46.1",
```
Comment (alebedev) [inline package.json:103]: This supports `root.$type` errors -- good abstraction for server errors that aren't field-specific

File: src/portals/hope/features/search/quotes/components/QuoteResult.tsx [hunk 2]
```diff
@@ -238,3 +249,30 @@ export const QuoteResult: React.FC<{
...
+}) => {
+  const isSignable =
+    quote.state === 'QUOTED' && quote.createdAt && !isExpired(quote)
+  console.log(quote?.state, isSignable, quote?.createdAt)
+
+  return (
+    <>
+      <Flex gap="tiny">
+        <Popup text="Debug info">
+          <code>{JSON.stringify(quote, null, 2)}</code>
```
Comment (falklinus) [inline src/portals/hope/features/search/quotes/components/QuoteResult.tsx:266]: I don't know the exact use case without any images, but we also have a PrettyPrintJSON component that might be of interest

File: src/portals/hope/features/search/quotes/components/QuoteResult.tsx [hunk 3]
```diff
@@ -238,3 +249,30 @@ export const QuoteResult: React.FC<{
...
+  quote: Pick<Quote, 'id' | 'startDate' | 'state' | 'createdAt'>
+}) => {
+  const isSignable =
+    quote.state === 'QUOTED' && quote.createdAt && !isExpired(quote)
+  console.log(quote?.state, isSignable, quote?.createdAt)
+
+  return (
+    <>
+      <Flex gap="tiny">
+        <Popup text="Debug info">
```
Comment (falklinus) [inline src/portals/hope/features/search/quotes/components/QuoteResult.tsx:265]: There is a component called `Modal` that you can use as dialog if that's the behaviour you're after

File: src/portals/hope/features/search/quotes/QuoteSign.tsx [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+  return (
+    <Wrapper>
+      <h3>⚠️ Danger zone ⚠️</h3>
+      <p>Manually sign quote {id}, bypassing all validation</p>
+      <form onSubmit={handleSubmit(submitHandler)} autoComplete="off">
+        <Input {...register('firstName')} type="text" label="First name" />
+        <Input {...register('lastName')} type="text" label="Last name" />
+        <Input {...register('email')} type="email" label="email" />
+        <Input
```
[the remark this answers is not in the document]
Comment (alebedev) [reply src/portals/hope/features/search/quotes/QuoteSign.tsx:55]: Sounds good, will do separate PR adapting it to react-hook-form

**Pull request closed without merging**